genesis/experiment/member/member.go

```diff
@@ -17,24 +17,82 @@
 package member
 
 import (
+	"github.com/insolar/insolar/core"
+	"github.com/insolar/insolar/cryptohelpers/ecdsa"
+	"github.com/insolar/insolar/genesis/experiment/member/signer"
+	"github.com/insolar/insolar/genesis/proxy/member"
+	"github.com/insolar/insolar/genesis/proxy/rootdomain"
+	"github.com/insolar/insolar/genesis/proxy/wallet"
 	"github.com/insolar/insolar/logicrunner/goplugin/foundation"
 )
 
 type Member struct {
 	foundation.BaseContract
 	Name      string
-	PublicKey []byte
+	PublicKey string
 }
 
 func (m *Member) GetName() string {
 	return m.Name
 }
-func (m *Member) GetPublicKey() []byte {
+func (m *Member) GetPublicKey() string {
 	return m.PublicKey
 }
 
-func New(name string) *Member {
+func New(name string, key string) *Member {
 	return &Member{
-		Name: name,
+		Name:      name,
+		PublicKey: key,
 	}
 }
+
+func (m *Member) AuthorizedCall(ref string, method string, params []interface{}, seed []byte, sign []byte) ([]interface{}, *foundation.Error) {
+	serialized, err := signer.Serialize(ref, method, params, seed)
+	if err != nil {
+		return nil, &foundation.Error{S: err.Error()}
+	}
+	verified, err := ecdsa.Verify(serialized, sign, m.PublicKey)
+	if err != nil {
+		return nil, &foundation.Error{S: err.Error()}
+	}
+	if !verified {
+		return nil, &foundation.Error{S: "Incorrect signature"}
+	}
+
+	switch method {
+	case "CreateMember":
```

Comment:
What do you think about private handler method for each case in this function?   
case "A":
    aHandler(ref, params)
case "B":
   ....